src/serializers/decoders/current/v10/decode_graph_schema.c

```diff
@@ -0,0 +1,87 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "decode_v10.h"
+
+static Schema *_RdbLoadSchema(RedisModuleIO *rdb, SchemaType type, bool already_loaded) {
+	/* Format:
+	 * id
+	 * name
+	 * #indices
+	 * (index type, indexed property) X M */
+
+	int id = RedisModule_LoadUnsigned(rdb);
+	char *name = RedisModule_LoadStringBuffer(rdb, NULL);
+	Schema *s = already_loaded ? NULL : Schema_New(name, id);
+	RedisModule_Free(name);
+
+	Index *idx = NULL;
+	uint index_count = RedisModule_LoadUnsigned(rdb);
+	for(uint i = 0; i < index_count; i++) {
+		IndexType type = RedisModule_LoadUnsigned(rdb);
+		char *field = RedisModule_LoadStringBuffer(rdb, NULL);
+
+		if(!already_loaded)
+			Schema_AddIndex(&idx, s, field, type);
+		RedisModule_Free(field);
+	}
+	if(s) {
+		if(s->index) Index_Construct(s->index);
```

Comment:
Add comment: `no entities are expected to be in the graph in this point in time`
consider asserting this statement